### DISCUSSION TITLE
Webpack loader and "new Worker" syntax

### DISCUSSION THREAD

**Discussion by JCarlesVilaseca** [gildas-lormeau/zip.js#229] [Q&A] [answered]
Used in Angular 10 application it needs to use another loader.

I would appreciate any help.

https://github.com/gildas-lormeau/zip.js/blob/27663ae2a7dc592cab9dfe63b0d05161c9523901/lib/core/codecs/codec-pool-worker.js#L80

```
ERROR in .../node_modules/@zip.js/zip.js/lib/core/codecs/codec-pool-worker.js 80:70
Module parse failed: Unexpected token (80:70)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
|       let messageTask;
|       if (!workerData.interface) {
>               workerData.worker = new Worker(new URL(workerData.scripts[0], import.meta.url));
|               workerData.worker.addEventListener(MESSAGE_EVENT_TYPE, onMessage, false);
|               workerData.interface = {
```

**Comment by gildas-lormeau** [comment]
Unfortunately, I'm not an Angular specialist. If you know how to integrate [this loader](https://www.npmjs.com/package/@open-wc/webpack-import-meta-loader) then it should work, I guess. You can find more info about this issue in Webpack 4 here: https://github.com/gildas-lormeau/zip.js/issues/212#issuecomment-769766135.

**Comment by gildas-lormeau** [comment, marked as answer]
Here is the procedure to integrate `webpack-import-meta-loader` in Angular 10 (copied from https://github.com/gildas-lormeau/zip.js/issues/191#issuecomment-763650835).

1. Install the custom webpack builder: `npm install -D @angular-builders/custom-webpack`
2. Install the plugin to load metadata: `npm install -D @open-wc/webpack-import-meta-loader`
3. Edit your `angular.json` file:
```json
"architect": {
      "build": {
          "builder": "@angular-builders/custom-webpack:browser",
          "options": {
            "customWebpackConfig": {
              "path": "./custom-webpack.config.js"
            },
  ...
      "serve": {
          "builder": "@angular-builders/custom-webpack:dev-server",
  ...
```
4. Add a `custom-webpack.config.js` with the following content:
```javascript
const webpack = require('webpack');

module.exports = {
  module: {
    rules: [
      {
        test: /\.js$/,
        loader: require.resolve('@open-wc/webpack-import-meta-loader')
      }
    ]
  }
};

```

**Reply by bastyen** [reply]
```
"server": {
          "builder": "@angular-builders/custom-webpack:browser",
          "options": {
            "customWebpackConfig": {
              "path": "./custom-webpack.config.js"
            }
```
